crates/trigger-redis/src/lib.rs

```diff
@@ -99,7 +99,11 @@ impl TriggerExecutor for RedisTrigger {
             .collect();
 
         // wait for the first handle to be returned and drop the rest
-        let (_, _, rest) = futures::future::select_all(tasks).await;
+        let (result, _, rest) = futures::future::select_all(tasks).await;
+        if let Err(err) = result? {
+            eprintln!("Error: {err:?}");
```

Comment:
If you don't adopt the "return it from `run`" suggestion, you need to return an error or force an exit-unwind here with a non-zero exit code.